File: Easy/Python/2511. Maximum Enemy Forts That Can Be Captured.py
class Solution(object):
    def captureForts(self, forts):
        this_sum = 0
        count = 0
        this_max = 0

        for i in range(len(forts)):
            if forts[i] != 0:
                this_sum += forts[i]
                if this_sum == 0:
                    this_max = max(this_max, count)
                this_sum = forts[i]
                count = 0
            elif this_sum != 0:
                count += 1
        
        return this_max

File: Easy/JavaScript/2511. Maximum Enemy Forts That Can Be Captured.js
var captureForts = function(forts) {
    let sum = 0;
    let count = 0;
    let max = 0;

    for(let i = 0; i < forts.length; i++){
        if(forts[i] != 0){
            sum += forts[i];
            if(sum == 0)
                max = Math.max(max, count);
            sum = forts[i];
            count = 0;
        }else if(sum != 0)
            count++;
    }

    return max;
};
